app/src/main/java/bg/zahov/app/ui/settings/RadioSettingsDialog.kt

```diff
@@ -0,0 +1,60 @@
+package bg.zahov.app.ui.settings
+
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.wrapContentSize
+import androidx.compose.foundation.selection.selectable
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Card
+import androidx.compose.material3.RadioButton
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.window.Dialog
+
+@Composable
+fun <T> RadioSettingsDialog(
+    title: String,
+    items: List<T>,
+    onDismissRequest: () -> Unit,
+    selected: String,
+    setSelected: (selected: T) -> Unit
+) {
+    Dialog(onDismissRequest = { onDismissRequest() }) {
```

Comment:
you can directly assign callback lambda without calling it -  `onDismissRequest = onDismissRequest`